numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -328,28 +328,31 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 gentype_str(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
+    PyObject  *item, *item_str;
 
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
     }
-    return ret;
+
+    item_str = PyObject_Str(item);
+    Py_DECREF(item);
+    return item_str;
 }
```

Comment:
Ah! I a few days ago I was struggling to remember where I had seen an example user-defined type. Thanks for reminding me it is `rational`.

I might add some tests involving it.